support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,235 @@
+package org.apereo.cas.ticket.registry;
+
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import org.apereo.cas.serializer.TicketSerializer;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketCatalog;
+import org.apereo.cas.ticket.TicketDefinition;
+import org.apereo.cas.ticket.TicketState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * @author David Rodriguez
+ *
+ * @since 5.2.0
+ */
+public class CassandraTicketRegistry extends AbstractTicketRegistry {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraTicketRegistry.class);
+    private static final String LAST_RUN_TABLE = "ticket_cleaner_lastrun";
+    private static final String EXPIRY_TABLE = "ticket_cleaner";
+
+    private static final int FIRST_COLUMN_INDEX = 0;
+    private static final long TEN_SECONDS = 10000L;
+    private static final int TEN = 10;
+    private static final int TICKETS_IN_TESTS = 10;
+
+    private final TicketCatalog ticketCatalog;
+    private final TicketSerializer<String> serializer;
+
+    private final PreparedStatement insertTgtStmt;
+    private final PreparedStatement updateTgtStmt;
+    private final PreparedStatement selectTgtStmt;
+    private final PreparedStatement deleteTgtStmt;
+
+    private final PreparedStatement insertStStmt;
+    private final PreparedStatement updateStStmt;
+    private final PreparedStatement selectStStmt;
+    private final PreparedStatement deleteStStmt;
+
+    private final PreparedStatement selectExStmt;
+    private final PreparedStatement selectDateExStmt;
+
+    private final PreparedStatement selectLrStmt;
+    private final PreparedStatement updateLrStmt;
+
+    private final Session session;
+    private final String tgtTable;
+    private final String stTable;
+
+    public CassandraTicketRegistry(final Session session, final TicketCatalog ticketCatalog,
+                                   final TicketSerializer<String> serializer) {
+        this.ticketCatalog = ticketCatalog;
+        this.serializer = serializer;
+
+        this.session = session;
+
+        tgtTable = ticketCatalog.find("TGT").getProperties().getStorageName();
```

Comment:
None of this should happen. The registry must be able to support all kinds of ticket types; not just TGTs and STs. You will need to create queries, schemas and everything else dynamically based on the catalog never ever assuming anything about the ticket type manually and rely entirely on the ticket catalog.

Reply:
@mmoayyed could you please point me to a ticket registry that uses tables and columns and need to differentiate tickets for storing/retrieving  them in a different way? 
I don't see how to remove ticket type checking when we need to store them in separate tables with different columns.

Reply:
You may be able to review MongoDb or DynamoDb ticket registries.

Reply:
Hello @mmoayyed and happy new year!
just had a look to the MongoDB and DynamoDb registries but still have a question. For that 2 registries the way the tickets are stored is the same (id, json) but for Cassandra we need the expiration date for TGTs (and similar) so we need a way to differentiate them.

probably we could have this different query saved within the ticketCatalog? how does it sound?

Reply:
Happy new year to you too!

Why do you need the expiration date for TGTs? You cannot or should not store the expiration date anywhere because it's not something that you can or should calculate statically.

Reply:
it's for making the cleaner a bit more smart, retrieving from DB just the possible expired tickets.
The cleaner will check anyway if the ticket is actually expired or not.

Otherwise, the cleaner will fetch ALL the tickets, right? for us this is not possible..

Reply:
Gotcha. 

No, you don't need to get all the tickets. You can design your own cleaner that constructs a very specific query to only retrieve tickets that are expired, rather than getting them all and checking them one by one for expiration. Based on the ticket expiration time, you can design a query with a where clause that looks for tickets whose creation time is greater than or less than blah, etc and then only process those and clean them.

Reply:
Also, please close this PR and start a new one. This is getting quite long, confusing and convoluted and it's difficult to keep track of notes and comments.

Reply:
correct me if I'm wrong,
- storing creationDate is not optimal as the `SELECT * WHERE creationDate < now - 20minutes` could return loads of tickets not expired.
- In the other hand, if you can store and update the ticket by expirationDate, the amount of tickets returned will be smaller. Sounds better, right? I don't see the drawbacks of this

Reply:
The drawbacks are that that expiration date is not a static value and cannot be calculated beforehand in most cases, especially if you want to account for idle timeouts or other dynamic forms of `TicketExpirationPolicy`. You might want to design this such that in your deployment, you override the necessary bits to make it work the way you need to until the expiration policy API is changed to make things behave more streamlined.